src/Microsoft.AspNetCore.Mvc.Razor/DependencyInjection/MvcRazorMvcCoreBuilderExtensions.cs

```diff
@@ -72,6 +72,11 @@ private static void AddRazorViewEngineFeatureProviders(IMvcCoreBuilder builder)
             {
                 builder.PartManager.FeatureProviders.Add(new MetadataReferenceFeatureProvider());
             }
+
+            if (!builder.PartManager.FeatureProviders.OfType<PrecompiledViewsFeatureProvider>().Any())
```

Comment:
Yup, that's the plan. Since it technically is very low cost discovery, the hope is that it doesn't have a significant startup cost. The gesture to precompile is entirely optional.